ruby-training/trainees/david/test_katas/08/lib/playing_card_deck.rb

```diff
@@ -4,10 +4,23 @@
 class PlayingCardDeck < Deck
   def initialize
     super
-    # your code here
+    @ranks = [*(2..10), 'J', 'Q', 'K', 'A']
```

Comment:
There's way too much code in this constructor. Avoid blocks, loops, conditionals and any complex code directly in the constructor. Refactor this stuff out into methods with methods with names that when read out loud look like steps that have to be followed.

It should look somewhat like this (use your own method names unless you like these a lot):

```ruby
def initialize
  super
  generate_ranks
  generate_suits
  populate_deck
  shuffle_deck
end
```